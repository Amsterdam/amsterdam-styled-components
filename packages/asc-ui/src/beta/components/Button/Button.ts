const Button = () => 'test'

export default Button
