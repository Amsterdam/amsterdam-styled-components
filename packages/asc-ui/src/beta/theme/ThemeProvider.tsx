import deepMerge from 'deepmerge'
import type { FunctionComponent, ReactNode } from 'react'
import { ThemeProvider as AscThemeProvider } from 'styled-components'
import { ascBetaTheme } from '.'
import type { Theme } from '../types'

export interface ThemeProviderProps {
  overrides?: Partial<Theme.ThemeInterface>
  theme?: Theme.ThemeInterface
  deep?: boolean
  children: ReactNode
}

const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  theme: defaultTheme = ascBetaTheme,
  overrides,
  deep = true,
  children,
}) => {
  const theme: Theme.ThemeInterface = deep
    ? deepMerge(defaultTheme, overrides || {})
    : { ...defaultTheme, ...overrides }

  return <AscThemeProvider theme={theme}>{children}</AscThemeProvider>
}

export default ThemeProvider
