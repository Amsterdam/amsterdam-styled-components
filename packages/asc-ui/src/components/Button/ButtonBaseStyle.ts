import { transitions } from 'polished'
import styled, { Theme } from '@datapunt/asc-core'
import { color as themeColor, focusStyleOutline } from '../../utils'
import { flexboxMinHeightFix } from '../shared/ie-fixes'

export type Props = {
  color?: Theme.TypeLevel
  square?: boolean
  hoverColor?: string
  href?: string
  size?: number
}

const ButtonBaseStyle = styled.button<Props>`
  display: inline-flex;
  align-items: center;
  text-decoration: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: normal;
  padding: 0 10px 0 10px;
  ${focusStyleOutline()}
  ${transitions(['color', 'background-color'], '0.1s ease-in-out')}
  background: ${({ color, theme }) => themeColor(color)({ theme })};

  &:hover {
    background: ${({ color, theme }) =>
      color
        ? themeColor(color, 'dark')({ theme })
        : themeColor('tint', 'level3')({ theme })};
  }

  ${flexboxMinHeightFix()}
`

export default ButtonBaseStyle
