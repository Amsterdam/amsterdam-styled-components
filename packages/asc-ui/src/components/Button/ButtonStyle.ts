import { darken, readableColor, size, transitions } from 'polished'
import styled, { css } from 'styled-components'
import { Theme } from '../../types'
import { svgFill, themeColor, themeSpacing } from '../../utils'
import Icon from '../Icon'
import { flexboxMinHeightFix } from '../shared/ie-fixes'

export type ButtonVariant =
  | 'primary'
  | 'secondary'
  | 'tertiary'
  | 'primaryInverted'
  | 'textButton'
  | 'blank'
  | 'application'

const defaultProps = {
  size: 30,
}

export const getButtonHeight = (theme: Theme.ThemeInterface) =>
  themeSpacing(11)({ theme })

export const ArrowRight = styled.div`
  position: absolute;
  top: 0;
  left: 100%;
  width: 0;
  height: 0;
  border: 22px solid rgba(255, 255, 255, 0);
  border-left: 15px solid ${themeColor('secondary')};
  border-right: 0;
  ${transitions('border-color', '0.1s ease-in-out')}
`

export const IconLeft = styled(Icon)`
  margin-right: 10px;
`
export const IconRight = styled(Icon)`
  margin-left: 10px;
`

const getVariant = () => ({
  theme,
  variant,
  color,
}: {
  theme: Theme.ThemeInterface
  variant?: ButtonVariant
  color?: Theme.ColorType
}) => {
  switch (variant) {
    case 'primary':
      return css`
        min-width: 90px;
        background-color: ${themeColor('primary')};
        color: ${readableColor(themeColor('primary')({ theme }))};
        ${svgFill(themeColor('tint', 'level1'))};

        &:focus,
        &:hover {
          background-color: ${darken(0.1, themeColor('primary')({ theme }))};
        }
      `

    case 'secondary':
      return css`
        min-width: 90px;
        background-color: ${themeColor('secondary')};
        color: ${themeColor('tint', 'level1')};
        ${svgFill(themeColor('tint', 'level1'))};

        &:focus,
        &:hover {
          background-color: ${darken(0.1, themeColor('secondary')({ theme }))};
        }

        ${(taskflow) =>
          taskflow &&
          css`
            &:not(:disabled)&:focus
              ${ArrowRight},
              &:not(:disabled)&:hover
              ${ArrowRight} {
              border-left-color: ${darken(
                0.1,
                themeColor('secondary')({ theme }),
              )};
            }
          `}
      `

    case 'tertiary':
      return css`
        min-width: 90px;
        background-color: ${themeColor('tint', 'level4')};
        ${svgFill(themeColor('tint', 'level7'))};

        &:focus,
        &:hover {
          background-color: ${darken(
            0.1,
            themeColor('tint', 'level4')({ theme }),
          )};
        }
      `

    case 'primaryInverted':
      return css`
        min-width: 90px;
        color: ${themeColor('primary')};
        border: 1px solid ${themeColor('primary')};
        background-color: ${themeColor('tint', 'level1')};
        ${svgFill(themeColor('primary', 'main'))};

        &:hover {
          outline: 1px solid ${themeColor('primary')};
        }
      `

    case 'textButton':
      return css`
        height: auto;
        padding: 0;
        align-self: baseline;
        white-space: normal;
        text-align: left;
        color: ${themeColor('primary')};
        background-color: rgba(0, 0, 0, 0);
        ${svgFill(themeColor('primary', 'main'))};

        /* remove transition because it's async with Icon */
        ${transitions('color', '0s')}

        &:hover {
          color: ${themeColor('secondary')};
          ${svgFill(themeColor('secondary', 'main'))};
        }

        ${IconLeft} {
          margin-right: ${themeSpacing(1)};
        }
        ${IconRight} {
          margin-left: ${themeSpacing(1)};
        }
      `

    case 'blank':
      return css`
        background-color: ${themeColor('tint', 'level1')};
        ${svgFill(themeColor('tint', 'level7'))};
        &:hover {
          background-color: ${themeColor('tint', 'level3')};
        }
      `
    case 'application':
      return css`
        border: 1px solid ${themeColor('tint', 'level7')};
        background-color: ${themeColor('tint', 'level1')};
        height: 32px;
        padding: ${themeSpacing(1, 2)};
        ${svgFill(themeColor('tint', 'level7'))};
        &:hover {
          background-color: ${themeColor('tint', 'level4')};
        }
      `
    default:
      // About-to-be-deprecated codeblock to support the 'color' prop
      return css`
        color: ${color
          ? readableColor(themeColor(color)({ theme }))
          : themeColor('primary')({ theme })};
        ${color &&
        css`
          background: ${themeColor(color)};
        `}

        ${!color &&
        css`
          border: 1px solid ${themeColor('primary')};
        `}

        &:hover {
          background: ${color
            ? themeColor(color, 'dark')({ theme })
            : themeColor('tint', 'level3')({ theme })};
          ${!color &&
          css`
            outline: 1px solid ${themeColor('primary')};
          `}
        }
      `
  }
}

export type ButtonStyleProps = {
  /**
   * @deprecated Use variant instead. Pass the theme-color.
   */
  color?: Theme.ColorType
  /**
   * @deprecated Use size to create a button with equal width and height
   */
  square?: boolean
  /**
   * Set an equal height and width
   */
  size?: number
  /**
   * A variant, usually different background-color and color of a button
   */
  variant?: ButtonVariant
  /**
   * Add narrow arrow on the right side of the secondary button
   */
  taskflow?: boolean
}

const ButtonStyle = styled.button<ButtonStyleProps>`
  height: ${({ theme }) => getButtonHeight(theme)};
  white-space: nowrap;
  display: inline-flex;
  align-items: center;
  text-decoration: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: 700;
  line-height: 20px;
  padding: ${({ size: sizeProp, theme }) =>
    sizeProp ? '0' : themeSpacing(3, 4)({ theme })};
  ${({ size: sizeProp, square }) =>
    (sizeProp || square) && // make the button square
    css`
      display: flex;
      align-items: center;
      justify-content: center;
      ${size(sizeProp || defaultProps.size)}// width and height
    `}
  ${transitions(['color', 'background-color'], '0.1s ease-in-out')}
  ${getVariant()}
  ${flexboxMinHeightFix()} // ie fix
  ${({ taskflow }) =>
    taskflow &&
    css`
      position: relative;
      z-index: 0;
      && {
        margin-right: 25px;
      }
      &:focus ${ArrowRight}:after {
        opacity: 1;
      }
    `}
  ${Icon} {
    flex-shrink: 0;
  }
  &:disabled {
    cursor: default;
    outline: none;
    border: none;
    color: ${themeColor('tint', 'level4')};
    background-color: ${themeColor('tint', 'level3')};
    ${svgFill(themeColor('tint', 'level4'))};
    text-decoration: none;
    ${({ taskflow }) =>
      taskflow &&
      css`
        ${ArrowRight} {
          border-left-color: ${themeColor('tint', 'level3')};
        }
      `}
    ${({ variant }) =>
      variant &&
      variant === 'textButton' &&
      css`
        background-color: rgba(0, 0, 0, 0);
      `}
  }
`

export default ButtonStyle
