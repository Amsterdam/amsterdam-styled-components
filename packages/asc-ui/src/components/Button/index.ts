export { ButtonProps, default } from './Button'
