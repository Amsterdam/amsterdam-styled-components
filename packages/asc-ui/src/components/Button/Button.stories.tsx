import styled from '@datapunt/asc-core'
import {
  Download,
  ChevronRight,
  ChevronLeft,
  Close,
  Facebook,
  Twitter,
  Linkedin,
  Email,
  Enlarge,
  Print,
} from '@datapunt/asc-assets'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import Button from './Button'
import ShareButton from './ShareButton'

const ButtonBar = styled.div<{}>`
  display: flex;
  height: 100px;

  & > * {
    margin-right: 5px;
  }
`

storiesOf('Atoms/Buttons', module)
  .addDecorator(storyFn => (
    <div style={{ padding: '40px 10px' }}>{storyFn()}</div>
  ))
  .add('[deprecated] variants', () => (
    <ButtonBar>
      <Button>Default button</Button>
      <Button color="primary">Primary button</Button>
      <Button color="secondary">Secondary button</Button>
    </ButtonBar>
  ))
  .add('variants', () => (
    <ButtonBar>
      <Button variant="primary">Default button</Button>
      <Button variant="application" iconLeft={<Enlarge />} iconSize={10}>
        Application button
      </Button>
      <Button
        variant="secondary"
        iconLeft={<Download />}
        iconRight={<ChevronRight />}
      >
        Download button
      </Button>
      <Button variant="secondary" taskflow>
        Taskflow button
      </Button>
      <Button variant="tertiary" iconRight={<ChevronRight />}>
        Download button
      </Button>
      <Button variant="primaryInverted" iconLeft={<ChevronRight />}>
        Download button
      </Button>
      <Button variant="textButton" iconLeft={<ChevronLeft />} iconSize={14}>
        Text button
      </Button>
    </ButtonBar>
  ))
  .add('variants - disabled', () => (
    <ButtonBar>
      <Button disabled variant="primary">
        Default button
      </Button>
      <Button
        disabled
        variant="secondary"
        iconLeft={<Download />}
        iconRight={<ChevronRight />}
      >
        Download button
      </Button>
      <Button disabled variant="tertiary" iconRight={<ChevronRight />}>
        Download button
      </Button>
      <Button disabled variant="primaryInverted" iconLeft={<ChevronRight />}>
        Download button
      </Button>
      <Button disabled variant="secondary" taskflow>
        Taskflow button
      </Button>
      <Button
        disabled
        variant="textButton"
        iconLeft={<ChevronLeft />}
        iconSize={14}
      >
        Text button
      </Button>
    </ButtonBar>
  ))
  .add('with only an icon', () => (
    <>
      <Button size={30} variant="blank" icon={<ChevronRight />} />
      <Button size={60} variant="blank" iconSize={40} icon={<Close />} />
    </>
  ))
  .add('social buttons', () => (
    <ButtonBar>
      <ShareButton hoverColor="#3b5999" iconSize={30}>
        <Facebook />
      </ShareButton>
      <ShareButton hoverColor="#55acee">
        <Twitter />
      </ShareButton>
      <ShareButton hoverColor="#0077B5">
        <Linkedin />
      </ShareButton>
      <ShareButton>
        <Email />
      </ShareButton>
      <ShareButton>
        <Print />
      </ShareButton>
    </ButtonBar>
  ))
