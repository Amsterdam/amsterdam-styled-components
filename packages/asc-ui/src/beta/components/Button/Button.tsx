import type { ElementType, ButtonHTMLAttributes } from 'react'
import { forwardRef } from 'react'
import ButtonStyle from './ButtonStyle'

export type ButtonProps = {
  as?: ElementType
  variant?: 'primary' | 'secondary' | 'tertiary' | 'blank'
  small?: boolean
} & ButtonHTMLAttributes<HTMLButtonElement>

const Button = forwardRef(
  ({ as = 'button', children, ...otherProps }: ButtonProps, ref) => (
    <ButtonStyle as={as} ref={ref} {...otherProps}>
      {children}
    </ButtonStyle>
  ),
)

export default Button
