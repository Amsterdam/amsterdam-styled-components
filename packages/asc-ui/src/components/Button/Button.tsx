import * as React from 'react'
import ButtonStyle, {
  IconLeft,
  IconRight,
  Props as ButtonStyleProps,
  ButtonVariants,
} from './ButtonStyle'
import Icon, { defaultProps as iconDefaultProps } from '../Icon/Icon'

export type Props = {
  as?: keyof JSX.IntrinsicElements | React.ComponentType<any>
  iconLeft?: React.ReactNode
  iconRight?: React.ReactNode
  icon?: React.ReactNode
  iconSize?: number
  $as?: any
} & ButtonStyleProps &
  React.ButtonHTMLAttributes<HTMLButtonElement>

export { ButtonVariants }

const Button: React.FC<Props> = ({
  children,
  iconLeft,
  iconRight,
  icon,
  iconSize,
  $as,
  ...otherProps
}): any => {
  const iconProps = {
    size: iconSize || iconDefaultProps.size,
  }
  return (
    <ButtonStyle {...otherProps} as={$as}>
      {iconLeft && <IconLeft {...iconProps}>{iconLeft}</IconLeft>}
      {icon ? <Icon {...iconProps}>{icon}</Icon> : children}
      {iconRight && <IconRight {...iconProps}>{iconRight}</IconRight>}
    </ButtonStyle>
  )
}

export default Button
