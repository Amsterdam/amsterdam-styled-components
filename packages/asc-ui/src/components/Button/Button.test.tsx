import { render } from '@testing-library/react'
import Button from './Button'
import ThemeProvider from '../../theme/ThemeProvider'

describe('Button', () => {
  it('should render the button', () => {
    const { container } = render(
      <ThemeProvider>
        <Button color="primary">Hello</Button>
      </ThemeProvider>,
    )
    expect(container.firstChild).toBeDefined()
  })
})
