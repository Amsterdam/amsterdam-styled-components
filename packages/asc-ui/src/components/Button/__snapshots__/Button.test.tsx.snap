// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled components should render buttons 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0px;
}

.c0 > * {
  margin-right: 5px;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: normal;
  padding: 0 10px 0 10px;
  -webkit-transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  background: #004699;
  padding: 0 10px 0 10px;
  min-height: 38px;
  line-height: 1em;
  color: #fff;
}

.c1:focus {
  outline-color: #FEC813;
  outline-style: solid;
  outline-width: medium;
}

.c1:hover {
  background: #00387a;
}

.c1:after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c1 svg {
  width: 30px;
  height: 30px;
}

.c1 svg rect,
.c1 svg polygon,
.c1 svg path {
  fill: #fff;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: normal;
  padding: 0 10px 0 10px;
  -webkit-transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  background: #ec0000;
  padding: 0 10px 0 10px;
  min-height: 38px;
  line-height: 1em;
  color: #fff;
}

.c2:focus {
  outline-color: #FEC813;
  outline-style: solid;
  outline-width: medium;
}

.c2:hover {
  background: #bc0000;
}

.c2:after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c2 svg {
  width: 30px;
  height: 30px;
}

.c2 svg rect,
.c2 svg polygon,
.c2 svg path {
  fill: #fff;
}

<ThemeProvider>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "bright": Object {
            "main": "#ffffff",
          },
          "error": Object {
            "main": "#ec0000",
          },
          "primary": Object {
            "dark": "#00387a",
            "main": "#004699",
          },
          "secondary": Object {
            "dark": "#bc0000",
            "main": "#ec0000",
          },
          "support": Object {
            "focus": "#FEC813",
            "invalid": "#EC0000",
            "valid": "#00A03C",
          },
          "tint": Object {
            "level1": "#ffffff",
            "level2": "#F5F5F5",
            "level3": "#E6E6E6",
            "level4": "#B4B4B4",
            "level5": "#767676",
            "level6": "#323232",
            "level7": "#000000",
          },
        },
        "globalStyle": "",
        "typography": Object {
          "a": Object {
            "color": "#000000",
            "fontFamily": "inherit",
            "fontSize": "1rem",
            "fontWeight": 300,
            "letterSpacing": "-0.01562em",
            "lineHeight": 1,
          },
          "fontFamily": "AvenirNextLTW01-Regular, verdana, sans-serif",
          "fontSize": "16px",
          "fontWeightHeavy": 900,
          "fontWeightLight": 300,
          "fontWeightMedium": 500,
          "fontWeightRegular": 400,
          "h1": Object {
            "color": "#000000",
            "fontFamily": "inherit",
            "fontSize": "6rem",
            "fontWeight": 300,
            "letterSpacing": "-0.01562em",
            "lineHeight": 1,
          },
          "h2": Object {
            "color": "#000000",
            "fontFamily": "inherit",
            "fontSize": "6rem",
            "fontWeight": 300,
            "letterSpacing": "-0.01562em",
            "lineHeight": 1,
          },
          "h3": Object {
            "color": "#000000",
            "fontFamily": "inherit",
            "fontSize": "6rem",
            "fontWeight": 300,
            "letterSpacing": "-0.01562em",
            "lineHeight": 1,
          },
          "h4": Object {
            "color": "#000000",
            "fontFamily": "inherit",
            "fontSize": Array [
              "
      ",
              "1.125rem",
              "
    ",
            ],
            "fontWeight": 300,
            "letterSpacing": "-0.01562em",
            "lineHeight": 1,
          },
          "h5": Object {
            "color": "#000000",
            "fontFamily": "inherit",
            "fontSize": Array [
              "
      ",
              "1.125rem",
              "
    ",
            ],
            "fontWeight": 300,
            "letterSpacing": "-0.01562em",
            "lineHeight": 1,
          },
          "h6": Object {
            "color": "#000000",
            "fontFamily": "inherit",
            "fontSize": "6rem",
            "fontWeight": 300,
            "letterSpacing": "-0.01562em",
            "lineHeight": 1,
          },
          "p": Object {
            "color": "#000000",
            "fontFamily": "inherit",
            "fontSize": "1rem",
            "fontWeight": 300,
            "letterSpacing": "-0.01562em",
            "lineHeight": 1.25,
          },
        },
      }
    }
  >
    <ButtonBar>
      <styled.div
        padding={0}
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-chPdSV",
                "isStatic": false,
                "lastClassName": "c0",
                "rules": Array [
                  "
  display: flex;
  ",
                  [Function],
                  "

  & > * {
    margin-right: 5px;
  }
",
                ],
              },
              "defaultProps": Object {
                "padding": 5,
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-chPdSV",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          padding={0}
        >
          <div
            className="c0"
          >
            <Button
              color="primary"
            >
              <Styled(styled.button)
                color="primary"
              >
                <StyledComponent
                  color="primary"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-kGXeez",
                        "isStatic": false,
                        "lastClassName": "c2",
                        "rules": Array [
                          "
  display: inline-flex;
  align-items: center;
  text-decoration: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: normal;
  padding: 0 10px 0 10px;
  ",
                          [Function],
                          "
  ",
                          "transition: color 0.1s ease-in-out, background-color 0.1s ease-in-out;",
                          ";

  background: ",
                          [Function],
                          ";

  &:hover {
    background: ",
                          [Function],
                          ";
  }

  ",
                          "
  &:after {
    content: '';
    display: block;
    min-height: inherit;
    font-size: 0;
  }
",
                          "
",
                          "
  padding: 0 10px 0 10px;
  min-height: 38px;
  line-height: 1em
  color: ",
                          [Function],
                          ";

  ",
                          [Function],
                          "

  &:hover {
    ",
                          [Function],
                          "
  }

  & svg {
    width: 30px;
    height: 30px;
    ",
                          [Function],
                          ";
  }
",
                        ],
                      },
                      "displayName": "Styled(styled.button)",
                      "foldedComponentIds": Array [
                        "sc-jzJRlG",
                      ],
                      "render": [Function],
                      "styledComponentId": "sc-kGXeez",
                      "target": "button",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <button
                    className="c1"
                    color="primary"
                  >
                    Hello
                  </button>
                </StyledComponent>
              </Styled(styled.button)>
            </Button>
            <Button
              color="secondary"
            >
              <Styled(styled.button)
                color="secondary"
              >
                <StyledComponent
                  color="secondary"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-kGXeez",
                        "isStatic": false,
                        "lastClassName": "c2",
                        "rules": Array [
                          "
  display: inline-flex;
  align-items: center;
  text-decoration: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: normal;
  padding: 0 10px 0 10px;
  ",
                          [Function],
                          "
  ",
                          "transition: color 0.1s ease-in-out, background-color 0.1s ease-in-out;",
                          ";

  background: ",
                          [Function],
                          ";

  &:hover {
    background: ",
                          [Function],
                          ";
  }

  ",
                          "
  &:after {
    content: '';
    display: block;
    min-height: inherit;
    font-size: 0;
  }
",
                          "
",
                          "
  padding: 0 10px 0 10px;
  min-height: 38px;
  line-height: 1em
  color: ",
                          [Function],
                          ";

  ",
                          [Function],
                          "

  &:hover {
    ",
                          [Function],
                          "
  }

  & svg {
    width: 30px;
    height: 30px;
    ",
                          [Function],
                          ";
  }
",
                        ],
                      },
                      "displayName": "Styled(styled.button)",
                      "foldedComponentIds": Array [
                        "sc-jzJRlG",
                      ],
                      "render": [Function],
                      "styledComponentId": "sc-kGXeez",
                      "target": "button",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <button
                    className="c2"
                    color="secondary"
                  >
                    World
                  </button>
                </StyledComponent>
              </Styled(styled.button)>
            </Button>
          </div>
        </StyledComponent>
      </styled.div>
    </ButtonBar>
  </ThemeProvider>
</ThemeProvider>
`;
